packages/component/src/providers/CustomElements/customElements/CodeBlockCopyButton.tsx

```diff
@@ -23,37 +24,45 @@ const CodeBlockCopyButton = memo(
     const [pressed, setPressed] = useState(false);
     const valueRef = useRefFrom(value);
 
-    const handleClick = useCallback(() => {
-      if (disabledRef.current) {
-        return;
-      }
+    const handleClick = useCallback<MouseEventHandler<HTMLButtonElement>>(
+      event => {
+        if (disabledRef.current) {
+          return;
+        }
+
+        let obsoleted = false;
 
-      let obsoleted = false;
+        (async () => {
+          try {
+            const parentElement = (event.currentTarget as HTMLButtonElement)?.parentElement?.parentElement;
 
-      (async () => {
-        try {
-          const { state } = await navigator.permissions.query({ name: 'clipboard-write' as any });
+            const { state } = await navigator.permissions.query({ name: 'clipboard-write' as any });
 
-          if (!obsoleted) {
-            if (state === 'granted') {
-              await navigator.clipboard?.write([
-                new ClipboardItem({ 'text/plain': new Blob([valueRef.current], { type: 'text/plain' }) })
-              ]);
+            const value =
```

Comment:
Curious, I guess it should always render as `<CodeBlockCopyButton value="text to copy">`. Assuming the DOM structure is kind of weird.

Reply:
We discussed something about fetching copied content from html instead of relying on the value, so this is my attempt on doing so for my use case.

As an alternative, we could listen for copy button clicks on the parent and expose a method on the component to copy the code. As the component is a React component, not sure it's better.

Reply:
Maybe follow this `copy` event by dispatching the copy event and allow parents to set `dataTransfer`?

https://developer.mozilla.org/en-US/docs/Web/API/Element/copy_event